File: .tmp_py4013/batch_04_sources/2526_find_consecutive_integers_from_a_data_stream.js
// LeetCode 2526 - Find Consecutive Integers from a Data Stream
// https://leetcode.com/problems/find-consecutive-integers-from-a-data-stream/

/**
 * @param {number} value
 * @param {number} k
 */
var DataStream = function(value, k) {
    this.value = value;
    this.k = k;
    this.streak = 0;
};

/** 
 * @param {number} num
 * @return {boolean}
 */
DataStream.prototype.consec = function(num) {
    if (num === this.value) this.streak++;
    else this.streak = 0;
    return this.streak >= this.k;
};

File: .tmp_rs4013/rust_check09/2526_find_consecutive_integers_from_a_data_stream.rs
// LeetCode 2526 - Find Consecutive Integers from a Data Stream
// https://leetcode.com/problems/find-consecutive-integers-from-a-data-stream/

pub struct DataStream {
    value: i32,
    k: i32,
    streak: i32,
}

impl DataStream {
    pub fn new(value: i32, k: i32) -> Self {
        Self {
            value,
            k,
            streak: 0,
        }
    }

    pub fn consec(&mut self, num: i32) -> bool {
        if num == self.value {
            self.streak += 1;
        } else {
            self.streak = 0;
        }
        self.streak >= self.k
    }
}

fn main() {}
